File: src/main/java/no/nav/bidrag/beregn/samvaersfradrag/SamvaersfradragCore.java
package no.nav.bidrag.beregn.samvaersfradrag;

import no.nav.bidrag.beregn.samvaersfradrag.dto.BeregnSamvaersfradragGrunnlagCore;
import no.nav.bidrag.beregn.samvaersfradrag.dto.BeregnetSamvaersfradragResultatCore;
import no.nav.bidrag.beregn.samvaersfradrag.periode.SamvaersfradragPeriode;

public interface SamvaersfradragCore {

  BeregnetSamvaersfradragResultatCore beregnSamvaersfradrag(BeregnSamvaersfradragGrunnlagCore beregnSamvaersfradragGrunnlagCore);

  static SamvaersfradragCore getInstance() {
    return new SamvaersfradragCoreImpl(SamvaersfradragPeriode.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/samvaersfradrag/SamvaersfradragCore.kt
package no.nav.bidrag.beregn.samvaersfradrag

import no.nav.bidrag.beregn.samvaersfradrag.dto.BeregnSamvaersfradragGrunnlagCore
import no.nav.bidrag.beregn.samvaersfradrag.dto.BeregnetSamvaersfradragResultatCore
import no.nav.bidrag.beregn.samvaersfradrag.periode.SamvaersfradragPeriode

fun interface SamvaersfradragCore {
    fun beregnSamvaersfradrag(grunnlag: BeregnSamvaersfradragGrunnlagCore): BeregnetSamvaersfradragResultatCore

    companion object {
        fun getInstance(): SamvaersfradragCore = SamvaersfradragCoreImpl(SamvaersfradragPeriode.getInstance())
    }
}
